src/js/index.d.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * A list of routes and their parameters and bindings.
+ *
+ * Extended and filled by the route list generated automatically by Ziggy.
+ */
+export interface RouteList {
+    //
+}
+
+/**
+ * A route name registered with Ziggy.
+ */
+type KnownRouteName = keyof RouteList;
+
+/**
+ * A route name, or any string.
+ */
+type RouteName = KnownRouteName | (string & {});
```

Comment:
I answered my own earlier question on the original PR here by playing around with it—`string` makes autocompletion stop working, I guess because it's too wide a type, but `(string & {})` still supports autocompletion while also allowing any old string. Does this functionality have a name? Something about it being 'exhaustive'?

Reply:
No, I also thought very long about it but did not find why this works. I just found that somewhere on SO :)

Reply:
Hahaa okay, thanks! Any thoughts about my other two questions below?

Reply:
Sorry, there some beautiful changes at home having my attention right now, so I am a bit slower in answering currently :)

Reply:
No worries at all, I really appreciate these answers it's super helpful ❤️